server/src/modules/post/post.service.ts

```diff
@@ -343,20 +316,12 @@ export class PostService {
           'answer_count',
         ],
       ],
-    })
+    })) as PostWithUserRelations
```

Comment:
Reason for typing as PostWithUserRelation is because `authService.verifyUserCanViewPost` accepts only PostWithUserRelation in the controller function that calls this func.